### PR TITLE
refactor(ffi): cleanup streaming and json decoding

### PR DESCRIPTION
**Pull request by erka**

### CODE REVIEW

File: flipt-engine-ffi/src/http.rs [hunk 1]
```diff
@@ -623,10 +584,15 @@ mod tests {
...
-
+        let (tx, mut rx) = mpsc::channel(4);
+        let result = fetcher.handle_streaming(&tx).await;
         assert!(result.is_ok());
         mock.assert();
+        assert!(rx.len() == 2);
+        // check first result
+        let result = rx.recv().await.unwrap().unwrap();
+        assert_eq!("default", result.namespace.key);
+        assert_eq!(0, result.flags.len());
```
[the remark this answers is not in the document]
Comment (markphelps) [reply flipt-engine-ffi/src/http.rs:595]: The mock data is actually wrong, it should be like:

```
result: {
  namespaces: [
    { 
       namespace: { key: x },
       flags: [ ... ]
    }
  ]
}
```

[the remark this answers is not in the document]
Comment (erka) [reply flipt-engine-ffi/src/http.rs:607]: FYI `StreamResult` was a ` map` so I changed it to a `vec` per your example.

[the remark this answers is not in the document]
Comment (GeorgeMac) [reply flipt-engine-ffi/src/http.rs]: Actually, `namespaces` is a map in the real protocol.
Sorry for the confusion! We do intend to get the protocol openened up at some point.

```
result: {
  namespaces: {
    "x": { 
       namespace: { key: x },
       flags: [ ... ]
    }
  }
}
```

[the remark this answers is not in the document]
Comment (erka) [reply flipt-engine-ffi/src/http.rs]: Thank you @GeorgeMac. I've reverted that change and updated the test.

[the remark this answers is not in the document]
Comment (GeorgeMac) [reply flipt-engine-ffi/src/http.rs]: Nice one!

**Review by markphelps — APPROVED** [review]
looks great! ty!

**Pull request merged**